xcp_d/utils/confounds.py

```diff
@@ -245,21 +253,30 @@ def load_acompcor(confounds_df, confoundjs):
             # Pull out variance explained for CSF masks that are retained
             if value["Mask"] == "CSF" and value["Retained"]:
                 CSF.append([key, value["VarianceExplained"]])
-    # Select the first five components and add them to the list
-    csflist = []
-    wmlist = []
-    for i in range(0, 4):
-        try:
-            csflist.append(CSF[i][0])
-        except Exception as exc:
-            pass
-            print(exc)
-        try:
-            wmlist.append(WM[i][0])
-        except Exception as exc:
-            pass
-            print(exc)
-    acompcor = wmlist + csflist
+
+    # grab up to 5 acompcor values
+    N_COLS_TO_GRAB = 5
+    wm_comp_cor_retained = WM
+    csf_comp_cor_retained = CSF
```

Comment:
Replace WM and CSF above with wm_comp_cor_retained and csf_comp_cor_retained, respectively.